internal/pkg/deploy/cloudformation/cloudformation.go

```diff
@@ -149,3 +164,21 @@ func stackExists(err error) bool {
 func envStackName(env *archer.Environment) string {
 	return fmt.Sprintf("%s-%s", env.Project, env.Name)
 }
+
+func getPrincipalArn() (string, error) {
+	sess, err := archersess.Default()
+
+	if err != nil {
+		return "", fmt.Errorf("failed to create default session: %w", err)
+	}
+
+	client := sts.New(sess)
+
+	identity, err := client.GetCallerIdentity()
```

Comment:
Can we do the same thing as `SSM` here where you store the sts client as a field in the struct?

This way I believe you won't have to add `var getPrincipalArnFunc = getPrincipalArn` for unit testing

Reply:
I avoided putting the STS client as a field on the `CloudFormation` struct because the session we need when initializing the STS client is different from the session we need when initializing the CF client.

The scope of where the STS client is needed is also very limited compared to the CF client which is used throughout the `cloudformation` package. Thoughts?

Reply:
yeah I see the struggle, I don't think there is an elegant solution either way but I'd lean towards the alternative above.

When we have the global variable `getPrincipalArnFunc` and replace its implementation in our unit tests we defeat the purpose of writing the test. Since the real code path is not walked.   

The alternative is storing a private field that's only accessed for `DeployEnvironment` but we at least get to invoke the real `getPrincipalArn()` method.

Reply:
Oh and also unit tests become more flaky too with a pointer to a function. 
1. You need to make sure to restore the original function after the test so that another test doesn't suddenly fail. 
2. You can't run your test cases in parallel.